aiida_vasp/io/outcar.py

```diff
@@ -0,0 +1,63 @@
+"""
+Tools for parsing OUTCAR files
+"""
+
+
+class OutcarParser(object):
+    """
+    parse OUTCAR into a dictionary, which is supposed to be turned into ParameterData later.
+    """
+
+    def __init__(self, fname):
+        super(OutcarParser, self).__init__()
+        self.outcar_file = fname
+        self.properties = ['volume', 'energies', 'efermi']
+
+    @property
+    def output_dict(self):
+        """Parse the OUTCAR file and return the parsed values wrapped in a dictionary"""
+        output_dict = {}
+        for property_name in self.properties:
+            try:
+                # call the method corresponding to property_name.
+                result = getattr(self, '_read_' + property_name)()
+                if result is not None:
+                    # read_property_name returns a dictionary if parsing has been successful.
+                    output_dict.update(result)
+            except AttributeError:
+                raise NotImplementedError('The OUTCAR parser does not implemnt _read_{}'.format(property_name))
+        return output_dict
+
+    def _read_volume(self):
+        """Parse the OUTCAR file and return the cell volume"""
+        result = {}
+        for line in open(self.outcar_file, 'r'):
```

Comment:
`open()` returns an open file object, either you should manually close it at the end of the function (guarantee it gets closed even if an exception is raised in between), or you can use `with open()` to do that for you:

```
with open(self.outcar_file, 'r') as outcar_file_obj:
  for line  in outcar_file_obj:
    ...
```